app/src/main/java/io/appium/uiautomator2/model/NotificationListener.java

```diff
@@ -1,114 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package io.appium.uiautomator2.model;
 
-import android.app.UiAutomation;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.view.accessibility.AccessibilityEvent;
 
 import java.util.ArrayList;
 import java.util.List;
 
-import io.appium.uiautomator2.core.UiAutomatorBridge;
+import io.appium.uiautomator2.core.UiAutomation;
 import io.appium.uiautomator2.utils.Logger;
 
+import static android.app.UiAutomation.OnAccessibilityEventListener;
 import static java.lang.System.currentTimeMillis;
 
+public final class NotificationListener implements OnAccessibilityEventListener {
+    private static final NotificationListener INSTANCE = new NotificationListener();
+    private static final int TOAST_CLEAR_TIMEOUT = 3500;
 
-public final class NotificationListener {
-    private final static NotificationListener INSTANCE = new NotificationListener();
-    private static List<CharSequence> toastMessages = new ArrayList<CharSequence>();
-    private final int TOAST_CLEAR_TIMEOUT = 3500;
-    private final int WAIT_FOR_EVENT_TIMEOUT = 500;
-    public boolean isListening = false;
-    private Listener listener;
+    private List<CharSequence> toastMessage = null;
+    private long previousTime = currentTimeMillis();
```

Comment:
what is the purpose of this property?

Reply:
it stores the time when the last toast was caught and used to manage toast's lifetime.

Reply:
then I'd rather rename it to something like recentToastTimestamp